runtime/shared_common/OSCachemmap.cpp

```diff
@@ -214,6 +215,9 @@ SH_OSCachemmap::startup(J9JavaVM* vm, const char* ctrlDirName, UDATA cacheDirPer
 	}
 	Trc_SHR_OSC_Mmap_startup_goodfileopen(_cachePathName, _fileHandle);
 
+	if (openMode & J9OSCACHE_OPEN_MODE_JITSERVER_AOT_LAYER) {
```

Comment:
The behaviour of this change looks like the same as the existing option [reset](https://eclipse.dev/openj9/docs/xshareclasses/#reset), which delete the top layer and recreate a new one. You can re-use the exiting code of `reset`.

Reply:
Actually, I'm not sure if the code handling the `reset` option be easily reused for this case, as that seems to be oriented toward destroying and then recreating the top layer, the reverse of what we want. Immediately deleting the underlying file during layer startup like this seems a little cleaner to me.

Reply:
OK. Then please add a trace point in this block.

Is there a reason not to use `SH_OSCachemmap::deleteCacheFile()` to delete the fie ?

Reply:
I don't believe there is. I'll use that.

Reply:
I'm assuming it won't be necessary to add a trace point to that block in that case, as `deleteCacheFile` has them already.

Reply:
I'm not sure how much sense it makes to include `lastErrorCode` handling and stopping of SCC startup if the file deletion fails, but that wouldn't be hard to add if it seems necessary.

Reply:
> I'm not sure how much sense it makes to include lastErrorCode handling and stopping of SCC startup if the file deletion fails, but that wouldn't be hard to add if it seems necessary.

What is the designed behaviour in this mode if the SCC file failed to be deleted ? If such error is non-fatal, we can continue to start up and consider issuing a warning to the user that the file failed to be deleted.

Reply:
> I'm assuming it won't be necessary to add a trace point to that block in that case, as deleteCacheFile has them already.

I would still prefer having a tracepoint which makes things more obvious for our service engineers in the future.

Reply:
> What is the designed behaviour in this mode if the SCC file failed to be deleted ? If such error is non-fatal, we can continue to start up and consider issuing a warning to the user that the file failed to be deleted.

I think it's non-fatal. The underlying file is unlinked because we don't want the temporary layer to leave any trace of itself behind on the file system. The layer should will still be usable if the file can't be deleted, so we might as well continue startup in that case. The only cleanup action I can think of would involve deleting the layer file, anyway.

Would a user message need a new NLS error for `errorHandler`?

> I would still prefer having a tracepoint which makes things more obvious for our service engineers in the future.

Sure, I'll add one.

Reply:
> What is the designed behaviour in this mode if the SCC file failed to be deleted ? If such error is non-fatal, we can continue to start up and consider issuing a warning to the user that the file failed to be deleted.

The common scenario we want to address is: JVM running in container with an embedded SCC that is marked read-only.
If we fail to delete the file there are 3 possible directions:
1. Make the SCC back to be read-only. I don't think that's possible, at least not without major code restructing
2. Fail the JVM.
3. Continue to run and give warning to user.

I think that option 3 is preferred here. If the SCC is embedded in the container, the file for the extra layer is going to be deleted when the container is destroyed. There is a possibility that the SCC is mapped to a volume and in this case the backing file for the extra layer will continue to exist past the container life-time (this is not a common solution since volumes may be shared across different nodes in Kubernetes and we don't allow remote SCC). I am not aware of a mechanism to detect whether a file is stored into persistent or ephemeral storage.

Reply:
> Would a user message need a new NLS error for errorHandler?

Yes.

Reply:
I'm not sure how those errors are created. It looks like they need translation as well? Does it have to be done in this PR, or can it be left for later? 

Displaying a message would certainly be good, if only to alert the user that they may need to figure out how to delete the extra layer themselves (somewhat unlikely, as @mpirvu points out).